Arm/Insts/DPI/Bitfield.lean

```diff
@@ -0,0 +1,83 @@
+/-
+Copyright (c) 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author(s): Shilpi Goel, Nevine
+-/
+-- For now, support only UBFM (immediate) aliased as LSR (immediate):
+-- 32- and 64-bit versions
+
+import Arm.Decode
+import Arm.Insts.Common
+
+namespace DPI
+
+open Std.BitVec
+
+@[simp]
+def exec_bitfield (inst: Bitfield_cls) (s : ArmState) : ArmState :=
+  -- TODO: match opc
+  let immr5 := inst.immr >>> 5
+  let imms5 := inst.imms >>> 5
+  if (inst.sf == 1 && inst.N != 1) ||
+     (inst.sf == 0 && (inst.N != 0 || immr5 != 0 || imms5 != 0)) then
+    write_err (StateError.Illegal s!"Illegal {inst} encountered!") s
+  else
+    let datasize  := if inst.sf == 1#1 then 64 else 32
+    let wtmask    := decode_bit_masks inst.N inst.imms inst.immr false datasize
+    match wtmask with
+    | none => write_err (StateError.Illegal s!"Illegal {inst} encountered!") s
+    | some (wmask, tmask) =>
+      let src := read_gpr datasize inst.Rn s
```

Comment:
We need to use `read_gpr_zr` when register 31 is mapped to the zero register ZR. See:
https://github.com/leanprover/LNSym/blob/main/Arm/State.lean#L317
and 
https://stackoverflow.com/questions/61532867/in-arm64-assembly-code-when-is-register-31-xzr-versus-sp